modules/pubsub/main.tf

```diff
@@ -67,22 +67,22 @@ resource "google_pubsub_topic_iam_member" "pubsub_sink_member" {
 # Pub/Sub topic subscription (for integrations) #
 #-----------------------------------------------#
 resource "google_service_account" "pubsub_subscriber" {
-  count        = var.create_subscriber ? 1 : 0
+  count        = var.create_subscriber_sa ? 1 : 0
   account_id   = local.subscriber_id
   display_name = "${local.topic_name} Topic Subscriber"
   project      = var.project_id
 }
 
 resource "google_pubsub_subscription_iam_member" "pubsub_subscriber_role" {
-  count        = var.create_subscriber ? 1 : 0
+  count        = var.create_subscriber_sa ? 1 : 0
   role         = "roles/pubsub.subscriber"
   project      = var.project_id
   subscription = local.pubsub_subscription
   member       = "serviceAccount:${google_service_account.pubsub_subscriber[0].email}"
 }
 
 resource "google_pubsub_topic_iam_member" "pubsub_viewer_role" {
-  count   = var.create_subscriber ? 1 : 0
+  count   = var.create_subscriber_sa ? 1 : 0
```

Comment:
Could you elaborate on why you need to break this functionality into two flags?

Reply:
Sure, so not all the SIEM tool requires service account to authenticate to the Pub/Sub, in few tools it uses users oauth tokens to authenticate. Though the existing SA creation is helpful, it might be additional/unnecessary resource for few deployments, hence the additional flag

Reply:
@sumeetoc in those cases I think the best approach would be to create the subscription outside of the module

Reply:
@bharathkkb are you suggesting to revert back the changes on pub/sub and if such requirement comes in then create subscription separately?

Reply:
@sumeetoc yes correct since I think its easy enough to create just the subscription use this module topic output for your usecase.

Reply:
@bharathkkb I've pushed the reverted changes for pub/sub, thanks for looking into this, please let me know if anything else is required